TestListPackages is flaky

Observed in https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/operator-framework_operator-registry/481/pull-ci-operator-framework-operator-registry-master-unit/1318308372407128064#1:build-log.txt%3A1492:

``` === RUN   TestListPackages
time="2020-10-19T21:55:10Z" level=info msg="loading Bundles" dir=../../manifests
time="2020-10-19T21:55:10Z" level=info msg=directory dir=../../manifests file=manifests load=bundles
time="2020-10-19T21:55:10Z" level=info msg=directory dir=../../manifests file=etcd load=bundles
time="2020-10-19T21:55:10Z" level=info msg=directory dir=../../manifests file=0.6.1 load=bundles
time="2020-10-19T21:55:10Z" level=info msg="found csv, loading bundle" dir=../../manifests file=etcdoperator.clusterserviceversion.yaml load=bundles
time="2020-10-19T21:55:10Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.6.1 file=etcdcluster.crd.yaml load=bundle name=etcdoperator.v0.6.1
time="2020-10-19T21:55:10Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.6.1 file=etcdoperator.clusterserviceversion.yaml load=bundle name=etcdoperator.v0.6.1
time="2020-10-19T21:55:12Z" level=info msg=directory dir=../../manifests file=0.9.2 load=bundles
time="2020-10-19T21:55:12Z" level=info msg="found csv, loading bundle" dir=../../manifests file=etcdoperator.v0.9.0.yaml load=bundles
time="2020-10-19T21:55:12Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdbackup.crd.yaml load=bundle name=etcdoperator.v0.9.0
time="2020-10-19T21:55:12Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdcluster.crd.yaml load=bundle name=etcdoperator.v0.9.0
time="2020-10-19T21:55:12Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdoperator.v0.9.0.yaml load=bundle name=etcdoperator.v0.9.0
time="2020-10-19T21:55:12Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdoperator.v0.9.2.clusterserviceversion.yaml load=bundle name=etcdoperator.v0.9.0
time="2020-10-19T21:55:12Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdrestore.crd.yaml load=bundle name=etcdoperator.v0.9.0
time="2020-10-19T21:55:14Z" level=info msg="found csv, loading bundle" dir=../../manifests file=etcdoperator.v0.9.2.clusterserviceversion.yaml load=bundles
time="2020-10-19T21:55:14Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdbackup.crd.yaml load=bundle name=etcdoperator.v0.9.2
time="2020-10-19T21:55:14Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdcluster.crd.yaml load=bundle name=etcdoperator.v0.9.2
time="2020-10-19T21:55:14Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdoperator.v0.9.0.yaml load=bundle name=etcdoperator.v0.9.2
time="2020-10-19T21:55:14Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdoperator.v0.9.2.clusterserviceversion.yaml load=bundle name=etcdoperator.v0.9.2
time="2020-10-19T21:55:14Z" level=info msg="loading bundle file" dir=../../manifests/etcd/0.9.2 file=etcdrestore.crd.yaml load=bundle name=etcdoperator.v0.9.2
time="2020-10-19T21:55:15Z" level=info msg=directory dir=../../manifests file=prometheus load=bundles
time="2020-10-19T21:55:15Z" level=info msg=directory dir=../../manifests file=0.14.0 load=bundles
time="2020-10-19T21:55:16Z" level=info msg="found csv, loading bundle" dir=../../manifests file=prometheusoperator.0.14.0.clusterserviceversion.yaml load=bundles
time="2020-10-19T21:55:16Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.14.0 file=alertmanager.crd.yaml load=bundle name=prometheusoperator.0.14.0
time="2020-10-19T21:55:16Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.14.0 file=prometheus.crd.yaml load=bundle name=prometheusoperator.0.14.0
time="2020-10-19T21:55:18Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.14.0 file=prometheusoperator.0.14.0.clusterserviceversion.yaml load=bundle name=prometheusoperator.0.14.0
time="2020-10-19T21:55:18Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.14.0 file=prometheusrule.crd.yaml load=bundle name=prometheusoperator.0.14.0
time="2020-10-19T21:55:18Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.14.0 file=servicemonitor.crd.yaml load=bundle name=prometheusoperator.0.14.0
time="2020-10-19T21:55:18Z" level=info msg=directory dir=../../manifests file=0.15.0 load=bundles
time="2020-10-19T21:55:20Z" level=info msg="found csv, loading bundle" dir=../../manifests file=prometheusoperator.0.15.0.clusterserviceversion.yaml load=bundles
time="2020-10-19T21:55:20Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.15.0 file=alertmanager.crd.yaml load=bundle name=prometheusoperator.0.15.0
time="2020-10-19T21:55:21Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.15.0 file=prometheus.crd.yaml load=bundle name=prometheusoperator.0.15.0
time="2020-10-19T21:55:22Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.15.0 file=prometheusoperator.0.15.0.clusterserviceversion.yaml load=bundle name=prometheusoperator.0.15.0
time="2020-10-19T21:55:22Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.15.0 file=prometheusrule.crd.yaml load=bundle name=prometheusoperator.0.15.0
time="2020-10-19T21:55:22Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.15.0 file=servicemonitor.crd.yaml load=bundle name=prometheusoperator.0.15.0
time="2020-10-19T21:55:23Z" level=info msg=directory dir=../../manifests file=0.22.2 load=bundles
time="2020-10-19T21:55:24Z" level=info msg="found csv, loading bundle" dir=../../manifests file=prometheusoperator.0.22.2.clusterserviceversion.yaml load=bundles
time="2020-10-19T21:55:24Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.22.2 file=alertmanager.crd.yaml load=bundle name=prometheusoperator.0.22.2
time="2020-10-19T21:55:24Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.22.2 file=prometheus.crd.yaml load=bundle name=prometheusoperator.0.22.2
time="2020-10-19T21:55:25Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.22.2 file=prometheusoperator.0.22.2.clusterserviceversion.yaml load=bundle name=prometheusoperator.0.22.2
time="2020-10-19T21:55:25Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.22.2 file=prometheusrule.crd.yaml load=bundle name=prometheusoperator.0.22.2
time="2020-10-19T21:55:25Z" level=info msg="loading bundle file" dir=../../manifests/prometheus/0.22.2 file=servicemonitor.crd.yaml load=bundle name=prometheusoperator.0.22.2
time="2020-10-19T21:55:26Z" level=info msg=directory dir=../../manifests file=strimzi-kafka-operator load=bundles
time="2020-10-19T21:55:27Z" level=info msg="found csv, loading bundle" dir=../../manifests file=strimzi-cluster-operator.v0.11.0.clusterserviceversion.yaml load=bundles
time="2020-10-19T21:55:27Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=Kafka-v1beta1.crd.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:27Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=KafkaBridge-v1alpha1.crd.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:27Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=KafkaConnect-v1beta1.crd.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:27Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=KafkaConnectS2I-v1beta1.crd.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:27Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=KafkaMirrorMaker-v1beta1.crd.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:27Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=KafkaTopic-v1beta1.crd.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:27Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=KafkaUser-v1beta1.crd.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:27Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=strimzi-cluster-operator.v0.11.0.clusterserviceversion.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:28Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=strimzi-cluster-operator.v0.11.1.clusterserviceversion.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:28Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=strimzi-cluster-operator.v0.12.1.clusterserviceversion.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:28Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=strimzi-cluster-operator.v0.12.2.clusterserviceversion.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
time="2020-10-19T21:55:28Z" level=info msg="loading bundle file" dir=../../manifests/strimzi-kafka-operator file=strimzi-kafka-operator.package.yaml load=bundle name=strimzi-cluster-operator.v0.11.0
--- FAIL: TestListPackages (20.01s)
    server_test.go:94: 
        	Error Trace:	server_test.go:94
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unavailable desc = timed out waiting for server handshake
        	Test:       	TestListPackages 
```

--- a/pkg/server/server_test.go
+++ b/pkg/server/server_test.go
@@ -26,12 +26,8 @@ const (
 	dbName  = "test.db"
 )
 
-func server() {
+func server(lis net.Listener) {
 	_ = os.Remove(dbName)
-	lis, err := net.Listen("tcp", port)
-	if err != nil {
-		logrus.Fatalf("failed to listen: %v", err)
-	}
 	s := grpc.NewServer()
 
 	db, err := sql.Open("sqlite3", dbName)
@@ -66,7 +62,11 @@ func server() {
 }
 
 func TestMain(m *testing.M) {
-	go server()
+	lis, err := net.Listen("tcp", port)
+	if err != nil {
+		logrus.Fatalf("failed to listen: %v", err)
+	}
+	go server(lis)
 	exit := m.Run()
 	if err := os.Remove(dbName); err != nil {
 		logrus.Fatalf("couldn't remove db")
@@ -841,7 +841,6 @@ func TestListBundles(t *testing.T) {
 		"strimzi-cluster-operator.v0.11.1",
 		"etcdoperator.v0.9.0",
 	}
-
 
 	var names []string
 	var gotBundles = make([]*api.Bundle, 0)

